File: app/src/main/java/com/niu/protect/manager/KeepAliveManger.java
package com.niu.protect.manager;

import android.app.Application;
import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import com.niu.protect.BabyApplication;
import com.niu.protect.keepalive.KeepAliveJobService;
import com.niu.protect.keepalive.LocalForegroundService;
import com.niu.protect.keepalive.RemoteForegroundService;
import com.niu.protect.service.FloatingService;
public class KeepAliveManger {
    private static KeepAliveManger instance;

    public static KeepAliveManger getInstance() {
        if (instance == null) {
            synchronized (KeepAliveManger.class) {
                if (instance == null) {
                    instance = new KeepAliveManger();
                }
            }
        }
        return instance;
    }

    public void keepAliveByTowService(Context context) {
        if (Build.VERSION.SDK_INT >= 28) {
            context.startForegroundService(new Intent(context, LocalForegroundService.class));
            context.startForegroundService(new Intent(context, RemoteForegroundService.class));
        } else {
            context.startService(new Intent(context, LocalForegroundService.class));
            context.startService(new Intent(context, RemoteForegroundService.class));
        }
        if (Build.VERSION.SDK_INT >= 21) {
            KeepAliveJobService.startJob(context);
        }
    }

    public void showNotification(Service service) {
        MineNotificationManager.getInstance().onCreate(service);
    }

    private void showOnePix() {
        Intent floatWinIntent = new Intent(BabyApplication.getInstance(), FloatingService.class);
        BabyApplication.getInstance().startService(floatWinIntent);
    }

    private void keepalive(Application context) {
    }
}

File: protect/src/main/java/com/niu/protect/manager/KeepAliveManger.kt
package com.niu.protect.manager

import android.app.Application
import android.app.Service
import android.content.Context
import android.content.Intent
import android.os.Build
import com.niu.protect.keepalive.KeepAliveJobService
import com.niu.protect.keepalive.LocalForegroundService
import com.niu.protect.keepalive.RemoteForegroundService
import com.niu.protect.service.FloatingService

/**
 * 保活
 *  */
class KeepAliveManger {
    fun keepAliveByTowService(context: Context) {
        context.startForegroundService(Intent(context, LocalForegroundService::class.java))
        context.startForegroundService(Intent(context, RemoteForegroundService::class.java))
        KeepAliveJobService.startJob(context)
    }

    fun showNotification(service: Service?) {
        MineNotificationManager.getInstance().onCreate(service)
    }

    private fun showOnePix(context: Context) {
        val floatWinIntent = Intent(context, FloatingService::class.java)
        context.startService(floatWinIntent)
    }

    private fun keepalive(context: Application) {}

    companion object {
        var instance: KeepAliveManger? = null
            get() {
                if (field == null) {
                    synchronized(KeepAliveManger::class.java) {
                        if (field == null) {
                            field = KeepAliveManger()
                        }
                    }
                }
                return field
            }
            private set
    }
}